src/main/java/com/drajer/bsa/ehr/service/impl/EhrFhirR4QueryServiceImpl.java

```diff
@@ -130,6 +121,41 @@
     return kd.getFhirInputData();
   }
 
+  public IGenericClient getClient(KarProcessingData kd, FhirContext context) {
+    String secret = kd.getHealthcareSetting().getClientSecret();
+    if (secret == null || secret.isEmpty()) {
+      backendAuthorizationService.getAuthorizationToken(kd);
+    } else {
+      ehrAuthorizationService.getAuthorizationToken(kd);
+    }
+
+    return fhirContextInitializer.createClient(
+        context,
+        kd.getHealthcareSetting().getFhirServerBaseURL(),
+        kd.getNotificationContext().getEhrAccessToken());
+  }
+
+  public void createResource(KarProcessingData kd, Resource resource) {
+
+    logger.info(" Getting FHIR Context for R4");
+    FhirContext context = fhirContextInitializer.getFhirContext(R4);
+
+    logger.info("Initializing FHIR Client");
+    IGenericClient client = getClient(kd, context);
+    client.create().resource(resource);
+  }
+
+  public void deleteResource(KarProcessingData kd, String id) {
+
+    logger.info(" Getting FHIR Context for R4");
+    FhirContext context = fhirContextInitializer.getFhirContext(R4);
+
+    IIdType iIdType = new IdType(id);
+    logger.info("Initializing FHIR Client");
+    IGenericClient client = getClient(kd, context);
+    client.delete().resourceById(iIdType);
```

Comment:
Same thing here for execute (though I haven't specifically tested this one, It should be the same)
```suggestion
    client.delete().resourceById(iIdType).execute();
```